pages/create-issue.tsx

```diff
@@ -1,23 +1,39 @@
 import { GetStaticProps } from 'next'
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
+import NumberFormat from 'react-number-format';
 import BeproService from '../services/bepro';
 import GithubMicroService from '../services/github-microservice';
 
-export default function PageCreateIssue() {
+interface Amount {
+  value?: string,
+  formattedValue: string,
+  floatValue?: number
+}
 
+export default function PageCreateIssue() {
 
   const [issueTitle, setIssueTitle] = useState("");
   const [issueDescription, setIssueDescription] = useState("");
-  const [issueAmount, setIssueAmount] = useState<number>(0);
+  const [issueAmount, setIssueAmount] = useState<Amount>({ value: '0', formattedValue: '0', floatValue: 0});
+  const [balance, setBalance] = useState<string>('0');
   const [allowedTransaction, setAllowedTransaction] = useState<boolean>(false);
 
+
+  useEffect(() => {
+    const resBepro = async () => {
+      await BeproService.login();
+      setBalance(await BeproService.network.getBEPROStaked())
```

Comment:
Good question. @moshmage "Is this `balance` state supposed to change at anytime, besides to catch it on component mount?"

Reply:
Create a helper for get "beprobalance" -> c7d5f37

Reply:
Not needed, bepro service is already as the service itself works as a "helper" (as in: we gain nothing by refactoring that part into a function)

Reply:
Can you clarify @moshmage where else, from a top-level, can we call `login()`? Because aparently it has to be called before every BeproService module for this to work. If this can't happen, we must create some custom hook, as I suggested, so `login()` will be already within and called just one time at one place.

Reply:
@soujvnunes we currently don't call it on a "top-level", as that's going to be handled by "My Account"; 

If the user needs a call to a `login()` statement somewhere before getting to this page, we should then create a navigation guard that checks for the logged in state _before_ the route is even shown - and if !login then we redirect the user to the "My Account" page where he can then connect/login against the metamask wallet and then be redirected back to where he was.


Again: A exported function for this sole propose is not needed nor should we have it as a dependency, we should use the service correctly instead, and use a navigational guard, and eventually account with the state from My Account

Reply:
Now I get it! @moshmage thank you!